build.gradle

```diff
@@ -0,0 +1,248 @@
+//           https://docs.gradle.org/current/userguide/task_configuration_avoidance.html
+//           need to avoid withType as well apparently?
+
+plugins {
+	id 'java'
+	id 'war'
+	id 'jacoco'
+	id 'org.ajoberstar.grgit' version '4.1.1'
+	id 'com.github.johnrengelman.shadow' version '8.1.1'
+}
+
+// TODO NOW test shadow jar works in groups - remove JNA - commit 2
+// TODO NOW get docserver working in WAR and test in docker-compose <- edit to start test server - commit 3
+// TODO NOW client jar - commit 4
+// TODO NOW client jar javadoc - needs java_common source - commit 4
+// TODO NOW schema updater script - commit 5
+// TODO NOW sdk-compile all, java, and docs - commit 6 
+// TODO NOW handle the git commit the same way as auth does - commit 7
+// TODO NOW delete build.xml, , and Makefile - commit 8
+// TODO NOW run tests from Eclipse w/o specifying classpath manually & remove sourceSets & claspath - commit 9
+
+repositories {
+	mavenCentral()
+}
+compileJava {
+	options.release = 11
+}
+
+tasks.withType(Test) {
+	/* 
+	 * TODO TEST Figure out why tests fail without this and remove. Might have something to do
+	 * with the stfuLoggers() call in many of the tests, might kill logging for tests that
+	 * require it
+	 * Although it seems to make Mongo start up correctly as well which is odd
+	 */
+	forkEvery = 1
+	/*
+	  * TODO TEST split tests into mongo wrapper tests & all other tests (incl. integration).
+	  * Set up GHA to run the non-mongo tests with a single version of mongo and run the
+	  * mongo tests with matrixed mongo versions. Combine coverage at the end somehow
+	*/
+	systemProperty "test.cfg", "./test.cfg"
+	maxHeapSize = "3G"
+	testLogging {
+		exceptionFormat = 'full'
+		showStandardStreams = true
+	}
+	filter {
+		// gradle thinks that classes annotated with @RunWith are tests
+		excludeTestsMatching "*Tester"
+	}
+}
+
+tasks.withType(JacocoReport) {
+	reports {
+		xml.required = true
+		csv.required = true
+	}
+}
+
+test {
+	finalizedBy jacocoTestReport
+}
+
+task testQuick(type: Test) {
+	// for Gradle 9.0 compatibility
+	testClassesDirs = testing.suites.test.sources.output.classesDirs
+	classpath = testing.suites.test.sources.runtimeClasspath
+
+	filter {
+		excludeTestsMatching "*LongTest"
+	}
+	finalizedBy "jacocoTestQuickReport" // must be a string, see TODOs at head of file
+}
+
+task jacocoTestQuickReport(type: JacocoReport, dependsOn: testQuick) {
+	sourceSets sourceSets.main
+	executionData(testQuick)
+}
+
+// Custom java project layout
+sourceSets {
+	main {
+		java {
+			srcDirs = ["src"]
+			exclude '**/test/**'
+		}
+	}
+	test {
+		java {
+			srcDirs = ["src"]
+			include '**/test/**'
+		}
+		resources {
+			srcDirs = ["src"]
+			include "**/*.properties"
+			include '**/*.spec'
+			include '**/*instance.*'
+			include '**/*.instance*'
+			include '**/*.html'
+			include '**/*.css'
+			include '**/*.gif'
+			include '**/*.js'
+			include '**/*.png'
+			include '**/*.txt'
+			include '**/*.weirdsuffix'
```

Comment:
Are these for future commits? There are not many files in the `test` folder

Reply:
These are all current files currently used in tests, but all the tests are under the src folder

Reply:
I assume they this part will be deleted once we move test folder out?

Reply:
yeah, I'll try to set things up like auth so we don't need the `sourceSets` section at all, but that comes last